### PR TITLE
add more ticks to the linear part of modified log scale

### LINKED ISSUE #2387: Modified log axis ticks are insufficient for the default domain
More ticks, please. :)

### PR DESCRIPTION
**Pull request by ztsai**
closes #2387

http://jsfiddle.net/ztsai/1zfz0tg2/1/

### CODE REVIEW

File: test/scales/modifiedLogScaleTests.ts [hunk 1]
```diff
@@ -190,5 +190,11 @@ describe("Scales", () => {
         assert.operator(ticks.length, ">", 0);
       });
     });
+
+    it("default domain has more than 2 ticks", () => {
+      scale.domain([200, -100]);
```
Comment (jtlan) [inline test/scales/modifiedLogScaleTests.ts]: This isn't the default domain, so we should remove this line.

**Comment by crmorford** [comment]
Probably unrelated, but do we know why do many of these tick labels aren't being drawn when they have plenty of room? http://jsfiddle.net/rtdxr3xv/5/

**Comment by crmorford** [comment]
The [10, -10] case has 3 ticks, but there's only room for one of them http://jsfiddle.net/rtdxr3xv/7/. 
Maybe it's out of scope for this, but it'd be awesome if it looked more like the [9, -9] case http://jsfiddle.net/rtdxr3xv/8/

**Comment by crmorford** [comment]
it looks really good on teeny axes though! http://jsfiddle.net/rtdxr3xv/9/

**Comment by bluong** [comment]
@crmorford 
Likely the above is unrelated, but it should definitely be addressed eventually.  The scope of this pull request is to see what ticks pop out using the default domain.  I think this means that you should be using only the default domain to test out the axes, in which case the [-10, 10] case is invalid as well as [9, -9].

That said, I do 100% agree that we should probably rethink our tick generation logic.  If you think the above makes sense, I think we should make an issue for the problems you're seeing and then adjust your process accordingly.

**Comment by jtlan** [comment]
@crmorford wrote:
>Probably unrelated, but do we know why do many of these tick labels aren't being drawn when they have plenty of room? http://jsfiddle.net/rtdxr3xv/5/

>Edit: something's weird, because on the reversed domain, a lot of ticks are drawn http://jsfiddle.net/rtdxr3xv/6/

Probably due to the same cause as #2385.

**Comment by bluong** [comment]
I take my comment back.

My comment is relevant if the scope of this pull request is to specifically solve the issue at #2387 .

However, based on the title of the pull request, the scope of the pull request encompasses more than just the relevant issue, also including cases that @crmorford has presented.

@ztsai if you are planning to just solve the relevant issue than adjust your pull request description title accordingly.  However, if you are expanding the scope to what is stated in the pull request title, proceed as normal.

**Pull request merged**